ui/src/LandingPage.js

```diff
@@ -335,6 +374,72 @@ class LandingPage extends Component {
     this.setState({ createPrintTemplateDialogDisplayed: false });
   };
 
+  onCreateSmsTemplate = async () => {
+    var failedValidation = false;
+
+    if (!this.state.packCode.trim()) {
+      this.setState({ packCodeValidationError: true });
+      failedValidation = true;
+    }
+    if (!this.state.notifyTemplateId) {
+      this.setState({
+        notifyTemplateIdValidationError: true,
+      });
+      failedValidation = true;
+    }
+    if (!this.state.template.trim()) {
+      this.setState({ templateValidationError: true });
+      failedValidation = true;
+    } else {
+      try {
+        const parsedJson = JSON.parse(this.state.template);
+        if (!Array.isArray(parsedJson)) {
+          this.setState({ templateValidationError: true });
+          failedValidation = true;
+        } else {
+          const validTemplateItems = ["__uac__", "__qid__"];
+          parsedJson.forEach((item) => {
+            if (!validTemplateItems.includes(item)) {
+              this.setState({ templateValidationError: true });
+              failedValidation = true;
+            }
+          });
+        }
+      } catch (err) {
+        this.setState({ templateValidationError: true });
+        failedValidation = true;
+      }
+    }
+
+    if (failedValidation) {
+      return;
+    }
+
+    const newSmsTemplate = {
+      notifyTemplateId: this.state.notifyTemplateId,
+      packCode: this.state.packCode,
+      template: JSON.parse(this.state.template),
+    };
+
+    const response = await fetch("/api/smsTemplates", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newSmsTemplate),
+    });
+    if (!response.ok) {
+      const errorMessage = await response.text();
+      this.setState({
+        createSmsTemplateError: errorMessage,
+        notifyTemplateIdValidationError: true,
+        templateValidationError: true,
+        packCodeValidationError: true,
```

Comment:
seems unlikely that the pack code is the reason the call to the backend is failing?